cartographer/pose_graph/node.h

```diff
@@ -19,6 +19,7 @@
 
 #include "cartographer/pose_graph/proto/node.pb.h"
 
+#include <array>
```

Comment:
it is probably not needed here. Only in derived classes.

Reply:
`AddParameterBlock` has `std::array` in its prototype, right?